src/lib/realmdb/__tests__/UserProfileStorage.js

```diff
@@ -0,0 +1,455 @@
+    const mobile = userProfileStorage.getProfileFieldDisplayValue('mobile')
+    expect(mobile).toEqual(userProfileStorage.profile.mobile.value)
+  })
+
+  it('should get walletAddress display value', () => {
+    const walletAddress = userProfileStorage.getProfileFieldDisplayValue('walletAddress')
+    expect(walletAddress).toEqual(userProfileStorage.profile.walletAddress.value)
+  })
+
+  it('should not get display value for invalid field', () => {
+    const value = userProfileStorage.getProfileFieldDisplayValue('abc')
+    expect(value).toEqual(undefined)
+  })
+
+  it('should get display profile', () => {
+    const displayProfile = userProfileStorage.getDisplayProfile()
+    const { profile } = userProfileStorage
+    Object.keys(displayProfile).forEach(key => {
+      if (typeof displayProfile[key] !== 'function') {
+        expect(displayProfile[key]).toEqual(profile[key].display)
+      }
+    })
+  })
+
+  it('should get private profile', () => {
+    const privateProfile = userProfileStorage.getPrivateProfile()
+    const { profile } = userProfileStorage
+    Object.keys(privateProfile).forEach(key => {
+      if (typeof privateProfile[key] !== 'function') {
+        expect(privateProfile[key]).toEqual(profile[key].value)
+      }
+    })
+  })
+
+  it('should get fields privacy', () => {
+    Object.keys(userProfileStorage.profile).forEach(key => {
+      expect(userProfileStorage.profile[key].privacy).toEqual(userProfileStorage.getFieldPrivacy(key))
+    })
+  })
+
+  it('should pass profile validation', async () => {
+    const { isValid, errors } = await userProfileStorage.validateProfile(profile)
+    expect(isValid).toBeTruthy()
+    expect(errors).toEqual({})
+  })
+
+  it('should fail profile validation with empty profile', async () => {
+    const { isValid, errors } = await userProfileStorage.validateProfile()
+    expect(isValid).toBeFalsy()
+    expect(errors).toEqual({})
+  })
+
+  it('should fail profile validation with email error', async () => {
+    const { isValid, errors } = await userProfileStorage.validateProfile({
+      avatar: '',
+      email: '',
+      fullName: '',
+      mnemonic: '',
+      username: '',
+      mobile: '',
+      walletAddress: '',
+      smallAvatar: '',
+    })
+
+    // console.log(errors)
+    expect(isValid).toBeFalsy()
+    expect(errors).toEqual(expect.objectContaining({ email: 'Unavailable email' }))
+    expect(errors).toEqual(expect.objectContaining({ mnemonic: 'Unavailable mnemonic' }))
+    expect(errors).toEqual(expect.objectContaining({ username: 'Unavailable username' }))
+    expect(errors).toEqual(expect.objectContaining({ mobile: 'Unavailable mobile' }))
+    expect(errors).toEqual(expect.objectContaining({ walletAddress: 'Unavailable walletAddress' }))
+  })
+
+  it('should set profile field to private', async () => {
+    await userProfileStorage.setProfileFieldPrivacy('email', 'private')
+    expect(userProfileStorage.getFieldPrivacy('email')).toEqual('private')
```

Comment:
private means the display will be '****'
i  have a feeling you alredy have a similar test